### DISCUSSION TITLE
Continuously failing on test_conv3.tcl after modifying gen_weigths.py

### DISCUSSION THREAD

**Discussion by LeatherE** [Xilinx/finn-hlslib#72] [Q&A]
Hi. 
I am currently trying to test the conv layer in this library. But when I tried to modify the value(kernel_dim, stride, input_precision, ifm_channels, ofm_channels, ifm_dimension, ofm_dimension, activation_precision, simd, pe, w_precision) in the gen_weigths.py. It often failed at the stage of Instrumenting C test bench. I wonder that if there has any relation between each value?

**Comment by fpjentzsch** [comment]
Hi,
which errors do you see? Usually, these HLS functions assume that all parameters "match" or divide evenly. Otherwise, the behavior might be undefined or execution will fail outright. For example:

- SIMD must divide ifm_channels
- PE must divide ofm_channels
- ofm_dimension must match the kernel_dim, stride, and ifm_dimension, e.g. for ifm_dim=5, kernel_dim=3, stride=1 ofm_dim will be 3

**Reply by LeatherE** [reply]
Hi, 
thanks for your answers. but i'm still facing the same error. below are my error message.
![未命名123](https://user-images.githubusercontent.com/37671697/148326277-8a014e39-af84-4c14-840e-6c4c042a077e.png)
![未命名456](https://user-images.githubusercontent.com/37671697/148326279-53b801f3-9192-4a9e-8106-b1fad4e78747.png)

my parameters are:
  - kernel_dim = 5
  - stride = 1
  - input_precision = 1
  - ifm_channels = 6
  - ofm_channels = 16
  - ifm_dimension = 14
  - ofm_dimension = 10
  - activation_precision = 16
  - expand = 1
  - simd = 3
  - pe = 2
  - w_precision = 1
  - mmv=1

thanks!

**Reply by LeatherE** [reply]
Hi, 

Perhaps you can give me one usable parameter combination for one random layer in CNV-w1a1 which you used in bnn-pynq/cnv_end2end_example. So that I can test the throughput of one convolution layer.

Thanks!
